### ISSUE TITLE
[Accessibility issues - Error message] Multiple issues

### ISSUE THREAD

**Issue by clairewu0218** [microsoft/vscode#189314]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.80.2
- OS Version: Windows_NT x64 10.0.22621

This is an accessibility issue. The tool I used to test is [Accessibility Insights for Windows](https://accessibilityinsights.io/docs/windows/overview/) along with systems narrator.

Steps to Reproduce:
1. Type something that can trigger error messages. 
2. Open the error message.
3. Use keyboard to navigate through the buttons of the error message
4. Use screen reader to read out the error message

- Issue1:
The 'Move to next problem' and 'move to previous problem' buttons are not accessible through keyboard while navigating using tab key. The root cause seems to be the `isKeyboardFocusable=false` attribute
![image](https://github.com/microsoft/vscode/assets/104450101/b5b69d98-e453-4bf8-ab61-96ce7a582a95)
- Issue2: 
Screen reader is not announcing the name for the controls under error message. Controls like 'Go to next problem', 'Go to Previous problem' and 'Close'. 
Note: There's this empty square element in the accessibility tree, may not be the cause of this issue, but this is not reasonable
![image](https://github.com/microsoft/vscode/assets/104450101/65302dc8-da72-4d1b-827b-8ab46b4bdf80)
- Issue3:
The narrator does not announce the error message information after navigating to the error message body using keyboard.

**Comment by VSCodeTriageBot** [comment]
Thanks for creating this issue! It looks like you may be using an old version of VS Code, the latest stable release is 1.80.2. Please try upgrading to the latest version and checking whether this issue remains.

Happy Coding!

**Comment by clairewu0218** [comment]
> Thanks for creating this issue! It looks like you may be using an old version of VS Code, the latest stable release is 1.80.2. Please try upgrading to the latest version and checking whether this issue remains.
> 
> Happy Coding!

Updated my version and tested. The issue persists.

**Comment by meganrogge** [comment]
Thanks for the issue

> The 'Move to next problem' and 'move to previous problem' buttons are not accessible through keyboard while navigating using tab key. The root cause seems to be the isKeyboardFocusable=false attribute

These are navigable via arrow keys - though it's odd because the right most button is focused first and it should be the left most button 

> Screen reader is not announcing the name for the controls under error message. Controls like 'Go to next problem', 'Go to Previous problem' and 'Close'.

These are announced for me when I use left and right arrow keys

**Comment by meganrogge** [comment]
> The narrator does not announce the error message information after navigating to the error message body using keyboard.

I can reproduce this and I think we should announce the problem when `view problem` is invoked - maybe with an `alert` 

cc @jooyoungseo and @rperez030

**Comment by sandeep-sen** [comment]
> @sandeep-sen I know you own the problems panel - do you also own the in editor problems view?

Unfortunately, I believe it's a case of mistaken identity :) , I am a PM on Azure SDK team.

**Comment by rperez030** [comment]
> > The narrator does not announce the error message information after navigating to the error message body using keyboard.


I strongly suggest not to use Accessibility Insights for Windows or Windows Narrator to test VS Code. Accessibility Insights for Windows is intended to be used to test native Windows applications, not web content  or Electron based apps like VS Code. In the case of Windows narrator, it only works natively with apps that support the UIA provider API, which is not turned on by default in Chromium. When there is no native UIA support Narrator uses a translation API which makes troubleshooting more difficult. I would limit testing to NVDA or JAWS which support IA1 and have mature support for Chromium based apps.

With that out of the way, I cannot reproduce the issue regarding error messages with NVDA or JAWS, but perhaps I'm not following the appropriate steps. if I open the problems pane (CTRL +Shift +M), the error receives keyboard focus and is automatically read by both NVDA and JAWS on Windows, and Voiceover on macOS. What is that I am missing?

**Comment by meganrogge** [comment]
Thanks @rperez030 for the insight here. The user is creating a problem in an editor then using f8 to open the inline problems view. I agree with you that the problems pane works well from an accessibility perspective.

**Comment by rperez030** [comment]
> The user is creating a problem in an editor then using f8 to open the inline problems view. I agree with you that the problems pane works well from an accessibility perspective.

But there is already an alert associated to the use of F8. If I press F8, an alert gets triggered, and cursor moves to that line.

**Comment by meganrogge** [comment]
I think you're right then and this is working as designed - perhaps is just an issue with the choice of screen reader

**Comment by clairewu0218** [comment]
> Thanks for the issue
> 
> > The 'Move to next problem' and 'move to previous problem' buttons are not accessible through keyboard while navigating using tab key. The root cause seems to be the isKeyboardFocusable=false attribute
> 
> These are navigable via arrow keys - though it's odd because the right most button is focused first and it should be the left most button
> 
> > Screen reader is not announcing the name for the controls under error message. Controls like 'Go to next problem', 'Go to Previous problem' and 'Close'.
> 
> These are announced for me when I use left and right arrow keys

Sorry for the late reply. Thanks for the infomation.
When I navigate to 'Close' button, it only announces 'Close' button. For users solely rely on the narrators, they won't be able to know there are two buttons to the left as it's not the normal left-to-right focus order. Is it possible to announces something like "Button 3 of 3" in addition to just 'Close' button, or make it focus on the left-most button first to make it truly accessible?

Thanks,
Claire

**Comment by meganrogge** [comment]
Yes, I am not sure why the last button is receiving tab focus. @sandy081 we should fix that

**Comment by rperez030** [comment]
> Sorry for the late reply. Thanks for the infomation. When I navigate to 'Close' button, it only announces 'Close' button. For users solely rely on the narrators, they won't be able to know there are two buttons to the left as it's not the normal left-to-right focus order. Is it possible to announces something like "Button 3 of 3" in addition to just 'Close' button, or make it focus on the left-most button first to make it truly accessible?


The container for the buttons has a role toolbar, and it is announced as such by screen readers. That constitute a signal to the user that the buttons should be navigated using the left and right arrow keys. I do agree though that the first button in the toolbar should be where keyboard focus lands when moving to the toolbar. @meganrogge I suspect the same problem exist on other areas in the editor.

Thanks a lot @clairewu0218 . As a blind screen reader user myself, I had no idea that the close button was the one to the right of the toolbar, especially because keyboard navigation wraps around in that control, so that you can effectively press the right arrow key and move forward.

**Comment by meganrogge** [comment]
It seems to be working in other areas around the workbench - like the terminal / editor toolbars. it's a bug if that's not the case 😄
